kernel/src/process.rs

```diff
@@ -283,9 +293,8 @@ pub trait Process {
     fn short_app_id(&self) -> ShortID;
 
     /// Returns the version number of the binary in this process, as specified
-    /// in a TBF Program Header; if the Userspace Binary only has a TBF Main
-    /// Header, returns 0.
-    fn binary_version(&self) -> u32;
+    /// in a TBF Program Header; if the binary has no version assigned, return [None]
+    fn binary_version(&self) -> Option<BinaryVersion>;
```

Comment:
Previously, 0 marked the absence of the binary version. This approach works, but `Option` is the Rust way of doing this in my opinion.

Using a `u32` for the binary version is too powerful: several operations defined on the integer type don't make sense, like addition or division. Or do they?

In the end, NonZeroU32 has been chosen as the underlying type of the new `BinaryVersion` type. This allows code generation optimizations. In my tests, the new version compiles exactly the same to the old code.